lib/cli-engine.js

```diff
@@ -573,7 +572,9 @@ CLIEngine.prototype = {
             if (shell.test("-f", pattern) && !ignoredPaths.contains(pattern)) {
                 executeOnFile(pattern);
             } else {
-                glob.sync(pattern, globOptions).forEach(executeOnFile);
+                glob.sync(pattern, globOptions).filter(function(filename) {
+                    return !ignoredPaths.contains(filename);
```

Comment:
It shouldn't affect performance in any demonstrative way. We're moving from O(n) to O(2n), which isn't that big of a deal.

Agree, though, better to move this into `executeFile()`.